### ISSUE TITLE
How is mAP50-95(P) calculated?

### ISSUE THREAD

**Issue by PING-An32** [ultralytics/ultralytics#23179]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Now I have understood that the meaning of mAP50-95(P) is to set the OKS value between 0.5 and 0.95, with a stride of 0.05, and there are a total of 10 values. Then, the average value is taken. Suppose there are multiple bounding boxes in the validation set images, and each box contains multiple key points. We use the model to perform inference on the validation set, and the result obtained is random (sorted by confidence). How does YOLO know that, for example, the eyes of the first person belong to the first person rather than the third person? Does YOLO first perform detection box matching before calculating mAP50-95(P), such as setting IoU = 0.5 to correspond to the bounding boxes?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @PING-An32, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (including metric interpretation like mAP50-95(P) for pose), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

This is an automated response 🤖 and an Ultralytics engineer will also assist soon 🛠️. In the meantime, to help us give the most accurate guidance, please share:
-Your `ultralytics` version (`pip show ultralytics`)
-Your OS, Python, and PyTorch versions
-The exact model used (e.g., YOLO11-pose variant) and command (Python or CLI)
-A snippet of your validation output/logs showing the metric line(s) you’re referring to
-If possible, a small example (1–3 images + labels) demonstrating the “person/keypoint association” concern, or a clear description of your annotation format

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Each set of keypoints is associated with a box because the YOLO Pose architecture is designed that way. Both boxes and keypoints are detected simultaneously by two different heads. Every anchor is uniquely associated with a box and corresponding keypoints.
